andela/lotus-ah
# #160451610 Admin List all users functionality

#### What does this PR do?
    This PR create routes for listing all users by an admin on the platform
#### Description of Task to be completed?
     1. Write test for the routes
     2. Create the routes 
     3. Refactored all profile route
     4. Create the controllers
#### How should this be manually tested?
- Start the app locally on port `3000`
- authenticate an admin
- Construct list users url as follows ```/api/v1/profiles```
- Make a `GET` request to the url
- You should get a response with the structure below:
```js
{
    "status": "success",
    "message": "User exists",
    "profiles": [
        {
            "email": "nondefyde@gmail.com",
            "firstname": null,
            "lastname": null,
            "bio": null,
            "imageUrl": null,
            "username": "DevOps",
            "id": 1
        },
        {
            "email": "kunleadekunle@gmail.com",
            "firstname": null,
            "lastname": null,
            "bio": null,
            "imageUrl": null,
            "username": "Backend",
            "id": 2
        }
    ],
   "profilesCount": 2
}
```
#### Any background context you want to provide?
      Gives an admin the ability to see all users
#### What are the relevant pivotal tracker stories?
      #160451610
#### Screenshots
<img width="1680" alt="screen shot 2018-10-19 at 3 23 06 pm" src="https://user-images.githubusercontent.com/43237059/47224393-51d01c00-d3b3-11e8-9cf1-ccaec8bc3efa.png">

#### Questions:
   None
